core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/WorkspaceApiService.scala

```diff
@@ -67,13 +67,16 @@ trait WorkspaceApiService extends UserInfoDirectives {
         } ~
         path("workspaces" / "id" / Segment) { workspaceId =>
           get {
-            parameterSeq { allParams =>
-              complete {
-                workspaceServiceConstructor(ctx).getWorkspaceById(workspaceId,
-                                                                  WorkspaceFieldSpecs.fromQueryParams(allParams,
-                                                                                                      "fields"
-                                                                  )
-                )
+            parameters("userProject".optional) { userProject =>
```

Comment:
I believe you can do:
```scala
parameters("userProject".as[GoogleProjectId].optional)
```
and skip the code later down which maps the String into GoogleProjectId.

I haven't tried this myself so it might not actually work!

Reply:
I did actually try that originally! It resulted in some sort of JSON parse error, I think because `GoogleProjectId` *has* a `value: String` as opposed to *being* a String... though I'm not quite proficient enough in Scala to know if that explanation makes sense.

Reply:
quite possible; not a big deal here. Feel free to ignore this comment.